src/lib.rs

```diff
@@ -563,3 +584,145 @@ impl Encoder for URIReserved {
         }
     }
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum IriSegmentKind {
+    Segment,
+    SegmentNoColons,
```

Comment:
Is `SegmentNoColons` really necessary?

Reply:
According to the spec, yes.

```
   isegment-nz-nc = 1*( iunreserved / pct-encoded / sub-delims
                        / "@" )
                  ; non-zero-length segment without any colon ":"
```

Used in `ipath-noscheme = isegment-nz-nc *( "/" isegment )`.

Reply:
I think I forgot a remark about the name of this type. The RFC uses the term "component" to designate each part of the IRI. I think it should be better to stick to it and rename this type to `IriComponent`, that way it will not be confusing to also have a variant named `Segment`.

Reply:
Remove IriComponent altogether as it was redundant. 😄